### PR TITLE
feat: add shared config maven project

### PR DESCRIPTION
**Pull request by chingor13**
The new parent pom contains the build config shared between all our releasable artifacts.

### CODE REVIEW

File: google-api-grpc/proto-google-cloud-monitoring-v3/src/main/java/google/monitoring/v3/DroppedLabelsOuterClass.java [hunk 1]
```diff
@@ -1,914 +0,0 @@
-// Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (BenWhitehead) [inline google-api-grpc/proto-google-cloud-monitoring-v3/src/main/java/google/monitoring/v3/DroppedLabelsOuterClass.java]: What this java file supposed to be deleted?

Comment (chingor13) [reply google-api-grpc/proto-google-cloud-monitoring-v3/src/main/java/google/monitoring/v3/DroppedLabelsOuterClass.java]: That file was erroneously generated a long time ago. It's missing `com.` in the package. The real version is https://github.com/googleapis/google-cloud-java/blob/master/google-api-grpc/proto-google-cloud-monitoring-v3/src/main/java/com/google/monitoring/v3/DroppedLabelsOuterClass.java

File: google-api-grpc/proto-google-cloud-os-login-v1/pom.xml [hunk 2]
```diff
@@ -28,4 +28,18 @@
...
   </dependencies>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-checkstyle-plugin</artifactId>
+          <configuration>
+            <excludes>com/google/cloud/oslogin/**</excludes>
```
[the remark this answers is not in the document]
Comment (chingor13) [reply google-api-grpc/proto-google-cloud-os-login-v1/pom.xml]: Good idea, there's already a `checkstyle.excludes` property to use.

File: versions.txt [hunk 3]
```diff
@@ -55,6 +55,7 @@ google-cloud-redis:0.102.0-alpha:0.102.1-alpha-SNAPSHOT
 google-cloud-resourcemanager:0.102.0-alpha:0.102.1-alpha-SNAPSHOT
 google-cloud-scheduler:1.7.0:1.7.1-SNAPSHOT
 google-cloud-securitycenter:0.102.0:0.102.1-SNAPSHOT
+google-cloud-shared-config:0.0.0:0.0.1-SNAPSHOT
```
Comment (kolea2) [inline versions.txt]: we're releasing this?

Comment (chingor13) [reply versions.txt]: I *think* we have to release it. It's like how we release google-api-grpc and google-cloud-client's as poms to maven.

We could also wait until the split repo to actually do this. Doing it now will mean it gets versioned everytime we release the monorepo. When we have a split repo available to move it to, we would stop versioning it here.

Comment (BenWhitehead) [reply versions.txt]: It does have to be released. Anything that is referenced as a parent is also loaded a use time.

**Comment by chingor13** [comment]
Let's hold off on this until the repo split.

**Comment by igorbernstein2** [comment]
Should google-cloud-testing/google-cloud-bigtable-emulator's parent also change?

**Comment by chingor13** [comment]
We will introduce this artifact in the new split repo.

**Pull request closed without merging**